data-prepper-plugins/event-json-codecs/src/test/java/org/opensearch/dataprepper/plugins/codec/event_json/EventJsonInputOutputCodecTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.codec.event_json;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.mock;
+import org.mockito.Mock;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+
+import java.io.ByteArrayOutputStream;
+import java.io.ByteArrayInputStream;
+
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.log.JacksonLog;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+public class EventJsonInputOutputCodecTest {
+    private static final Integer BYTEBUFFER_SIZE = 1024;
+
+    private ByteArrayOutputStream outputStream;
+
+    @Mock
+    private EventJsonOutputCodecConfig eventJsonOutputCodecConfig;
+
+    @Mock
+    private EventJsonInputCodecConfig eventJsonInputCodecConfig;
+
+    private EventJsonOutputCodec outputCodec;
+    private EventJsonInputCodec inputCodec;
+
+    @BeforeEach
+    public void setup() {
+        outputStream = new ByteArrayOutputStream(BYTEBUFFER_SIZE);
+        eventJsonInputCodecConfig = mock(EventJsonInputCodecConfig.class);
+    }
+
+    public EventJsonOutputCodec createOutputCodec() {
+        return new EventJsonOutputCodec(eventJsonOutputCodecConfig);
+    }
+
+    public EventJsonInputCodec createInputCodec() {
+        when(eventJsonInputCodecConfig.getOverrideTimeReceived()).thenReturn(false);
```

Comment:
Let's move this to setup and then add a test for when the override_time_received is true

Reply:
Modified as per the suggestion